documentation and tests for Packets

Riyria-was-taken opened the pull request.

Riyria-was-taken committed:
added documentation to Packet.h

Riyria-was-taken committed:
added documentation to Packet.h

Riyria-was-taken committed:
implemented Packet tests

Riyria-was-taken committed:
implemented Packet tests

Riyria-was-taken committed:
fixed Packet tests

Riyria-was-taken committed:
fixed Packet tests

PawelA committed:
moved receive from Echo to RawReceiver (#30)

* moved receive from Echo to RawReceiver

* added base class

* temporary

* name change

* small stuff

* header guards

* virtual and other fixes

* bring back api files

* removed old tests

* bugfix, start and stop moved to ctor/dtor

* initlalizer list in ctor

* formatting

* renames header guards

* restructured the constructor

* review changes

Lorak-mmk commented on libecho/tests/packet_tests.cpp:
```diff
@@ -0,0 +1,266 @@
+#include <gtest/gtest.h>
+#include <algorithm>
+#include <cassert>
```
Is cassert used here? If not, delete it

Lorak-mmk commented on libecho/tests/packet_tests.cpp:
```diff
@@ -0,0 +1,266 @@
+
+uint16_t SIZE = 3, NUM = 2137;
+const std::vector<Flag> SET_FLAGS{Flag::ACK, Flag::DMD, Flag::LPC}, UNSET_FLAGS{Flag::SYN, Flag::FIN, Flag::RST};
+const std::vector<uint8_t> header{0, 38, 0, 3, 8, 89}, data{13, 4, 55}, incorrect_header{10, 37, 0, 4, 1, 2, 70},
+    incorrect_data_small{1}, incorrect_data_big{12, 34, 56, 78}, incorrect_crc_format_small{12},
+    incorrect_crc_format_big{12, 13, 14, 16, 17}, incorrect_crc{1, 1, 1, 1};
```
I think it would be much more readable, if every variable was in distinct line.
```suggestion
const std::vector<uint8_t> header{0, 38, 0, 3, 8, 89},
                           data{13, 4, 55},
                           incorrect_header{10, 37, 0, 4, 1, 2, 70},
                           incorrect_data_small{1},
                           incorrect_data_big{12, 34, 56, 78},
                           incorrect_crc_format_small{12},
                           incorrect_crc_format_big{12, 13, 14, 16, 17},
                           incorrect_crc{1, 1, 1, 1};
```

Lorak-mmk commented on libecho/tests/packet_tests.cpp:
```diff
@@ -0,0 +1,266 @@
+#include "Packet.h"
+
+uint16_t SIZE = 3, NUM = 2137;
+const std::vector<Flag> SET_FLAGS{Flag::ACK, Flag::DMD, Flag::LPC}, UNSET_FLAGS{Flag::SYN, Flag::FIN, Flag::RST};
```
```suggestion
const std::vector<Flag> SET_FLAGS{Flag::ACK, Flag::DMD, Flag::LPC}, 
                        UNSET_FLAGS{Flag::SYN, Flag::FIN, Flag::RST};
```

Lorak-mmk commented on libecho/tests/packet_tests.cpp:
```diff
@@ -0,0 +1,266 @@
+    auto p = Packet::loadHeaderFromBytes(header);
+    ASSERT_EQ(SIZE, p.getSize());
+    ASSERT_EQ(NUM, p.getNumber());
+    for (auto f : SET_FLAGS)
+        ASSERT_TRUE(p.isSet(f));
+    for (auto f : UNSET_FLAGS)
+        ASSERT_FALSE(p.isSet(f));
```
Inconsistent style - later you use braces aroung single-line for, use them here too

Lorak-mmk commented on libecho/tests/packet_tests.cpp:
```diff
@@ -0,0 +1,266 @@
+        ASSERT_EQ(d[i], data[i]);
+    }
+
+    for (auto f : SET_FLAGS)
+        ASSERT_TRUE(p.isSet(f));
+    for (auto f : UNSET_FLAGS)
+        ASSERT_FALSE(p.isSet(f));
```
Braces around fors

Lorak-mmk commented on libecho/tests/packet_tests.cpp:
```diff
@@ -0,0 +1,266 @@
+
+    p = pb.getPacket();
+
+    for (auto f : SET_FLAGS)
+        ASSERT_TRUE(p.isSet(f));
+    for (auto f : UNSET_FLAGS)
+        ASSERT_FALSE(p.isSet(f));
```
Braces around fors

Lorak-mmk commented on libecho/tests/packet_tests.cpp:
```diff
@@ -0,0 +1,266 @@
+    ASSERT_EQ(s, incorrect_data_big.size());
+    ASSERT_EQ(s, d.size());
+
+    for (size_t i = 0; i < data.size(); i++)
+        ASSERT_EQ(incorrect_data_big[i], d[i]);
```
Braces around fors

Lorak-mmk commented on libecho/tests/packet_tests.cpp:
```diff
@@ -0,0 +1,266 @@
+TEST(tests_packet_builder, test_setFlag) {
+    auto pb = PacketBuilder();
+
+    for (auto f : SET_FLAGS)
+        pb.setFlag(f);
+
+    auto p = pb.getPacket();
+
+    for (auto f : SET_FLAGS)
+        ASSERT_TRUE(p.isSet(f));
+    for (auto f : UNSET_FLAGS)
+        ASSERT_FALSE(p.isSet(f));
+
+    for (auto f : SET_FLAGS)
+        pb.setFlag(f);
```
Braces around fors

Lorak-mmk commented on libecho/tests/packet_tests.cpp:
```diff
@@ -0,0 +1,266 @@
+    ASSERT_EQ(s, data.size());
+    ASSERT_EQ(s, d.size());
+
+    for (size_t i = 0; i < data.size(); i++)
+        ASSERT_EQ(data[i], d[i]);
```
Braces

Lorak-mmk commented on libecho/tests/packet_tests.cpp:
```diff
@@ -0,0 +1,266 @@
+    ASSERT_EQ(s, data.size());
+    ASSERT_EQ(s, d.size());
+
+    for (size_t i = 0; i < data.size(); i++)
+        ASSERT_EQ(data[i], d[i]);
```
Braces

Lorak-mmk reviewed (changes requested):
Braces around fors pls

Riyria-was-taken committed:
resolved PR comments

Riyria-was-taken committed:
resolved PR comments

Riyria-was-taken committed:
ISender interface, derived class BitSender (#29)

- implemented interface for data sending - ISender.h as well as one class fulfiling it - BitSender.*

- moved part of responsibilities (encoding) from IAudioConverter.h, BitAudioConverter.h and moved data sending implementation from Echo.c to ISender.h, BitSender.*

- changes in ISender.h, BitSender.*: changed send, added setters and getters, fixed clang-tidy warnings

- updated CMakeLists

Riyria-was-taken committed:
Merge branch 'dev-packet-docs-tests' of https://github.com/Lorak-mmk/EchoConnect into dev-packet-docs-tests
